bluepyopt/ephys/mechanisms.py

```diff
@@ -119,16 +119,16 @@ def instantiate_determinism(self, deterministic, icell, isec, sim):
                 short_secname = sim.neuron.h.secname(sec=isec).split('.')[-1]
                 for iseg in isec:
                     seg_name = '%s.%.19g' % (short_secname, iseg.x)
-                    sim.neuron.h.setdata_StochKv(iseg.x, sec=isec)
+                    exec("sim.neuron.h.setdata_" + self.prefix + "(iseg.x, sec=isec)")
```

Comment:
I try and not use exec whenever possible.

Does:
`getattr(sim.neuron.h, "setdata_" + self.prefix)(iseg.x, sec=isec)`
work?